### PR TITLE
nixos/*: even more options doc md conversions

### PR DESCRIPTION
**Pull request by pennae**
###### Description of changes

once more unto the breach. as usual the final commit is generated with tool assistance to make sure the generated `options.xml` from which all docs are built remains the same with that commit applied.

358 options the tool can see remain unchanged. there are plenty more internal options that can't be converted like this, and a *lot* of options in modules that use wrapper functions to create their options. this could be the final PR that uses a conversion tool.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/misc/gitlab.nix [hunk 1]
```diff
@@ -752,8 +751,7 @@ in {
         description = ''
           The number of worker processes Puma should spawn. This
           controls the amount of parallel Ruby code can be
-          executed. GitLab recommends <quote>Number of CPU cores -
-          1</quote>, but at least two.
+          executed. GitLab recommends <literal>Number of CPU cores - 1</literal>, but at least two.
```
Comment (jtojnar) [inline nixos/modules/services/misc/gitlab.nix:754]: This is weird. I did not realize the dash was supposed to be minus until reading the older version of GitLab installation requirements it appears to come from: https://docs.gitlab.com/13.12/ee/install/requirements.html

Not sure how to make it better, though.

Comment (K900) [reply nixos/modules/services/misc/gitlab.nix:754]: Maybe just reword it in a followup? Something like "one less than the number of CPU cores, but no less than two".

Comment (jtojnar) [reply nixos/modules/services/misc/gitlab.nix:754]: BTW, the formula is more complicated nowadays: https://docs.gitlab.com/ee/install/requirements.html#puma-workers

**Comment by pennae** [comment]
changed the quote replacement to use `“”` after all now on request.

File: nixos/modules/services/misc/gitlab.nix (hunk 1 above)
Comment (pennae) [reply nixos/modules/services/misc/gitlab.nix:754]: probably best to just refer to the documentation here instead of giving any concrete guidance. :/

**Review by jtojnar — APPROVED** [review]
Looks great.

**Pull request merged**